api/envoy/api/v2/core/protocol.proto

```diff
@@ -83,6 +84,9 @@ message Http1ProtocolOptions {
   // Describes how the keys for response headers should be formatted. By default, all header keys
   // are lower cased.
   HeaderKeyFormat header_key_format = 4;
+
+  // Enables trailers for HTTP1. By default the HTTP/1 codec drops proxied trailers.
+  bool enable_trailers = 5;
```

Comment:
Should we protect this with an API flag or a runtime flag guard? I feel that we do want to proxy these by default going forward, after this has some soak time.

Reply:
We've already discussed this at length and we landed on config guard with default off. HTTP/1.1 trailers are really poorly supported on the open internet and I think it's quite likely default on will break something. My vote is to keep it config guarded w/ default off but I know @alyssawilk and @PiotrSikora might have slightly different opinions on this.